### PR TITLE
Implement function to fetch box Metadata and fix installBox function

### LINKED ISSUE #61: parameter `boxName` must be url?
https://github.com/personium/personium-client-java/blob/a4a91d8467fc70bf35b17b4e5f2c022a999c24fb/src/main/java/io/personium/client/Cell.java#L492

### PR DESCRIPTION
**Pull request by yoh1496**
### function to fetch box Metadata 

For below API, I implemented getString in DavCollection class.
https://personium.io/docs/ja/apiref/303_Progress_of_Bar_File_Installation/

### fix installBox function

Parameter `boxName` in installBox function in Cell class receive only box URL.
And, when `boxName` is URL, it returns broken Box object.
Is there any testing?

closes #61 

### Added other features

- Support for recursive DELETE box API

### CODE REVIEW

File: src/main/java/io/personium/client/DavCollection.java [hunk 1]
```diff
@@ -269,6 +269,15 @@ public ServiceCollection service(String name) {
         return new ServiceCollection(this.accessor, UrlUtils.append(this.getPath(), name));
     }
 
+    /**
+     * Return self string.
+     * @return GET Response from this
+     * @throws DaoException EXception thrown
```
Comment (tochi-y) [inline src/main/java/io/personium/client/DavCollection.java]:
```suggestion
     * @throws DaoException Exception thrown
```

**Comment by dixonsiu** [comment]
Thank you @yoh1496! It is a great function that has been missing.

File: src/main/java/io/personium/client/Cell.java [hunk 2]
```diff
@@ -491,7 +491,7 @@ public String getClassName() {
      */
     public Box installBox(String boxName, InputStream barFile) throws DaoException {
         RestAdapter rest = (RestAdapter) RestAdapterFactory.create(this.accessor);
-        rest.mkcol(boxName, barFile, IRestAdapter.CONTENT_TYPE_ZIP);
+        rest.mkcol(this.box(boxName, null).url.toString(), barFile, IRestAdapter.CONTENT_TYPE_ZIP);
         /** if an error response is returned in HTTP, DaoException will be thrown. */
         return this.box(boxName, null);
```
Comment (tochi-y) [inline src/main/java/io/personium/client/Cell.java]: How about changing to the following?

```java
Box box = this.box(boxName, null);
rest.mkcol(box.url.toString(), barFile, IRestAdapter.CONTENT_TYPE_ZIP);
return box;
```

File: src/main/java/io/personium/client/DavCollection.java [hunk 3]
```diff
@@ -269,6 +269,15 @@ public ServiceCollection service(String name) {
         return new ServiceCollection(this.accessor, UrlUtils.append(this.getPath(), name));
     }
 
+    /**
+     * Return self string.
+     * @return GET Response from this
```
Comment (tochi-y) [inline src/main/java/io/personium/client/DavCollection.java]: Can you adapt the comments to fit the contents of this method, along with other getString comments?

File: src/main/java/io/personium/client/DavCollection.java [hunk 4]
```diff
@@ -357,7 +366,7 @@ public WebDAV getStringWebDAV(String pathValue) throws DaoException {
      * @throws DaoException Exception thrown
      */
     public WebDAV getStringWebDAV(String pathValue, String charset) throws DaoException {
-        String url = UrlUtils.append(this.getPath(), pathValue);
+        String url = !pathValue.equals("") ? UrlUtils.append(this.getPath(), pathValue) : this.getPath();
```
Comment (tochi-y) [inline src/main/java/io/personium/client/DavCollection.java]: Have you confirmed that pathValue will not be null?

File: src/main/java/io/personium/client/BoxManager.java [hunk 5]
```diff
@@ -110,4 +115,18 @@ public Box retrieve(String name) throws DaoException {
         return new Box(accessor, json, UrlUtils.append(accessor.getCurrentCell().getUrl(), name));
     }
 
+    /**
+     * This method performs recursively delete box
+     * @param boxName boxName to be delete
+     * @return PersoniumResponse 
+     * @throws DaoException Library Exception
+     */
+    public PersoniumResponse recursiveDelete(String boxName) throws DaoException {
```
Comment (tochi-y) [inline src/main/java/io/personium/client/BoxManager.java:124]: Please add the explanation of the added content to the Pull Request next time.

**Review by tochi-y — CHANGES_REQUESTED** [review]
@yoh1496 Thank you for your modification. Please check the comments.

File: src/main/java/io/personium/client/BoxManager.java (hunk 5 above)
Comment (yoh1496) [reply src/main/java/io/personium/client/BoxManager.java:124]: I'm sorry for lack of explanation.

**Comment by yoh1496** [comment]
I modified codes you pointed out.
Could you review this again?

File: src/main/java/io/personium/client/DavCollection.java [hunk 6]
```diff
@@ -269,6 +269,15 @@ public ServiceCollection service(String name) {
         return new ServiceCollection(this.accessor, UrlUtils.append(this.getPath(), name));
     }
 
+    /**
+     * This method requests the GET method for the DAV collection itself by calling its overloaded version.
+     * @return GET Response int string form
+     * @throws DaoException Exception thrown
+     */
+    public String getString() throws DaoException {
```
Comment (tochi-y) [inline src/main/java/io/personium/client/DavCollection.java:277]: I think it is better to add interfece `getMetaData` to the BoxManager class to retrieve box meta data, .

**Review by tochi-y — CHANGES_REQUESTED** [review]
@yoh1496 Could you implement getMetaData method?

**Comment by yoh1496** [comment]
I added getMetaData method in Box class.
Could you review this again?

**Review by tochi-y — APPROVED** [review]
Thank you for your modification!

**Pull request merged**